File: csharp/435-non-overlapping-intervals.cs
public class Solution {
    public int EraseOverlapIntervals(int[][] intervals) {
        var sortedIntervals = intervals.Clone() as int[][];
        Array.Sort(sortedIntervals, (a, b) => a[0] - b[0]);
        
        var result = 0;
        var prevEnd = sortedIntervals[0][1];
        for (var i = 1; i < sortedIntervals.Length; i++) {
            if (sortedIntervals[i][0] >= prevEnd) {
                prevEnd = sortedIntervals[i][1];
            } else {
                result++;
                prevEnd = Math.Min(prevEnd, sortedIntervals[i][1]);
            }
        }
        return result;
    }
}

File: cpp/435-non-overlapping-intervals.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end(), [](const auto& a, const auto& b) {
            return a[0] < b[0];
        });

        int result = 0;
        int prevEnd = intervals[0][1];
        for (int i = 1; i < intervals.size(); i++) {
            if (intervals[i][0] >= prevEnd) {
                prevEnd = intervals[i][1];
            } else {
                result++;
                prevEnd = min(intervals[i][1], prevEnd);
            }
        }
        return result;
    }
};